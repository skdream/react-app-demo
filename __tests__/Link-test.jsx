import React from 'react';
import TestUtils from 'react-dom/test-utils'
import Link from '../src/Link';

import renderer from 'react-test-renderer';

it('正确渲染', () => {
    const tree = renderer.create( 
    <Link page = "http://www.chebada.com" > 车巴达 </Link>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('没有设置page时渲染为一个锚点',() => {
    const tree = renderer.create(
        <Link>车巴达</Link>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('当hover时改变class',() => {
    const component = renderer.create(
        <Link page="http://www.chebada.com">车巴达</Link>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onMouseEnter();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    tree.props.onMouseLeave();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})