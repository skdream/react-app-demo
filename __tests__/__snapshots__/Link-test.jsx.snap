exports[`test 当hover时改变class 1`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  车巴达
</a>
`;

exports[`test 当hover时改变class 2`] = `
<a
  className="hovered"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  车巴达
</a>
`;

exports[`test 当hover时改变class 3`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  车巴达
</a>
`;

exports[`test 正确渲染 1`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
   车巴达 
</a>
`;

exports[`test 没有设置page时渲染为一个锚点 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}>
  车巴达
</a>
`;

exports[`当hover时改变class 1`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  车巴达
</a>
`;

exports[`当hover时改变class 2`] = `
<a
  className="hovered"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  车巴达
</a>
`;

exports[`当hover时改变class 3`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  车巴达
</a>
`;

exports[`正确渲染 1`] = `
<a
  className="normal"
  href="http://www.chebada.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   车巴达 
</a>
`;

exports[`没有设置page时渲染为一个锚点 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  车巴达
</a>
`;
